modules/pricer/src/main/java/com/opengamma/strata/pricer/fx/SmileDeltaTermStructureParameters.java

```diff
@@ -5,479 +5,202 @@
-    ArgChecker.isTrue(riskReversal.length == nbExp,
-        "Risk reversal length should be coherent with time to expiry length");
-    ArgChecker.isTrue(strangle.length == nbExp, "Strangle length should be coherent with time to expiry length");
-    ArgChecker.isTrue(riskReversal[0].length == delta.length,
-        "Risk reversal size should be coherent with time to delta length");
-    ArgChecker.isTrue(strangle[0].length == delta.length, "Strangle size should be coherent with time to delta length");
-    _volatilityTerm = new SmileDeltaParameters[nbExp];
-    for (int loopexp = 0; loopexp < nbExp; loopexp++) {
-      _volatilityTerm[loopexp] = SmileDeltaParameters.of(
-          timeToExpiry[loopexp],
-          atm[loopexp],
-          DoubleArray.copyOf(delta),
-          DoubleArray.copyOf(riskReversal[loopexp]),
-          DoubleArray.copyOf(strangle[loopexp]));
-    }
-    ArgChecker.isTrue(_volatilityTerm[0].getVolatility().size() > 1,
-        "Need more than one volatility value to perform interpolation");
-  }
-
-  /**
-   * Obtains a copy of this {@link SmileDeltaTermStructureParameters}. 
+   * The smile for the specified time is created by applying time interpolator and extrapolator. 
+   * The smile is obtained from the data by the given interpolator.
    * 
-   * @return the copy
-   */
-  public SmileDeltaTermStructureParameters copy() {
-    return new SmileDeltaTermStructureParameters(getName(), getVolatilityTerm(), getTimeInterpolator());
-  }
-
-  /**
-   * Get smile at a given time. The smile is described by the volatilities at a given delta. The smile is obtained from the data by the given interpolator.
-   * @param time The time to expiry.
-   * @return The smile.
+   * @param expiry  the time to expiry
+   * @return the smile
    */
-  public SmileDeltaParameters getSmileForTime(double time) {
-    int nbVol = _volatilityTerm[0].getVolatility().size();
-    int nbTime = _timeToExpiry.length;
+  default public SmileDeltaParameters getSmileForTime(double expiry) {
+    int nbVol = getNumberStrike();
+    int nbTime = getNumberSmiles();
     ArgChecker.isTrue(nbTime > 1, "Need more than one time value to perform interpolation");
     double[] volatilityT = new double[nbVol];
     for (int loopvol = 0; loopvol < nbVol; loopvol++) {
       double[] volDelta = new double[nbTime];
       for (int looptime = 0; looptime < nbTime; looptime++) {
-        volDelta[looptime] = _volatilityTerm[looptime].getVolatility().get(loopvol);
+        volDelta[looptime] = getVolatilityTerm().get(looptime).getVolatility().get(loopvol);
       }
-      Interpolator1DDataBundle interpData =
-          _timeInterpolator.getDataBundleFromSortedArrays(_timeToExpiry, volDelta);
-      volatilityT[loopvol] = _timeInterpolator.interpolate(interpData, time);
+      BoundCurveInterpolator bound = getTimeInterpolator().bind(
+          getTimeToExpiry(), DoubleArray.copyOf(volDelta), getTimeLeftExtrapolator(), getTimeRightExtrapolator());
```

Comment:
`ofUnsafe(volData)`
or even better, use DoubleArray.of(size, i -> ...) to create the object

Can also use `DoubleMatrix.ofUnsafe()` a few lines down. Worth checking other methods for this, as we don't want to copy arrays unnecessarily.